util/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerDbStorage.java

```diff
@@ -267,7 +268,39 @@ public UUID getReplacementGuid() {
 
   @Override
   public Composite getReplacementOwnerKey() {
-    return getCryptoService().getOwnerPublicKey(voucher);
+    List<PublicKey> certs = null;
+
+    String sql =
+        "SELECT KEYS FROM OWNER_CUSTOMERS WHERE CUSTOMER_ID =  "
+            + "(SELECT CUSTOMER_ID FROM TO2_DEVICES WHERE GUID = ?)";
+
+    try (Connection conn = dataSource.getConnection();
+        PreparedStatement pstmt = conn.prepareStatement(sql)) {
+      pstmt.setString(1, guid.toString());
+      try (ResultSet rs = pstmt.executeQuery()) {
+        if (rs.next()) {
+          certs = PemLoader.loadPublicKeys(rs.getString(1));
+        }
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+
+    int keyType =
+        getCryptoService().getPublicKeyType(getCryptoService().getDevicePublicKey(voucher));
```

Comment:
Please get the key type from getOwnerPublicKey(), since, internally getDevicePublicKey() fetches key from devcertchain, which is absent for EPID devices

Reply:
Done